### ISSUE TITLE
Language used for project project?

### ISSUE THREAD

**Issue by karan10** [1ppm/1ppmLog#107]
It would be good if we have language/framework used column in HOF table. Will be easy for others to check other projects based on language. Your views?

**Comment by Bemmu** [comment]
The focus should be on getting stuff done, not to get it done with cool tech. This also reminds me I failed to do my 1PPM in January while trying to learn new cool tech instead -_-

**Comment by karan10** [comment]
It is perfectly fine and don't required yet. At the end of year Language will play role to make repository projects more descriptive. It will be better to show that [here]( http://1ppm.club/hallOfFame).

**Comment by ggerhard** [comment]
Thanks for sharing your ideas and starting a discussion! The current system worked well to get started quickly, but it's very limited. E.g lots of people didn’t even add a state column :( 
I'm working on migrating everything to a database. That will allow to add columns easily and also relations with other tables. First feature would be a progress bar, so you can show how far the project is. Also allowing more/better project descriptions and a distinction between past and planned projects etc 

For language and frameworks, I was thinking of a tagging system. So you can add tags like "python Django JavaScript" but maybe some else does "arduino blog". What do you think?

**Comment by karan10** [comment]
Yes great. Progress bar is a great idea. For tagging system, then we'll be having lots of "tags" in "miscellaneous" tag if we make one. :/ Also not everyone is looking for tag and you said right 

> _lots of people didn’t even add a state column_

We can skip it. :)

**Comment by ggerhard** [comment]
I was thinking of free tags like on medium, ideally with autocompletion. User starts typing "J", system suggests "Java" "JavaScript" "JSF",.. Think that would be more flexible than fixed categories. Eventually the system could display popular tags, a tag cloud, trending tags for this month etc.

I'm still thinking about categories and groups - would it make sense to create user categories/groups like "Blogger", "Hacker", "(Hardware-) Maker",... to make it easier to find likeminded users?  Or is it better to stay project centered?  

I will build an early beta ASAP, looking forward to your feedback ;)

**Comment by karan10** [comment]
Tag suggestion is also good. We need to have all the possible tags which can relate to what user is typing, if tag is not present in our list, then we have to save that keyword in our tags list for future suggestion. How shall we get all the possible keyword tags?
Also you can check [this](https://stackoverflow.com/questions/5318580/how-does-github-figure-out-a-projects-language/).

For group and categories, it will become ambiguous if a user's January project is "creating and writing blog" and February project is "Network usage tracking software in Python". In which category user is supposed to belong?

I think if the tag system worked, we can identify likeminded users through tags.

p.s. we can stay project centered until there are more demands for such things. :)
